CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+cmake_minimum_required(VERSION 2.8)
+
+cmake_policy(SET CMP0048 NEW)
+
+project(safeside VERSION 0.1.0 LANGUAGES CXX)
+
+add_library(safeside demos/cache_sidechannel.cc demos/instr.cc)
+
+# Spectre V1 -- bounds check bypass
+add_executable(spectre_v1 demos/spectre_v1.cc)
+target_link_libraries(spectre_v1 safeside)
+
+# Spectre V3 / Meltdown
+add_executable(meltdown demos/meltdown.cc)
+target_link_libraries(meltdown safeside)
+
+# Spectre V4 -- speculative store bypass
+add_executable(spectre_v4 demos/spectre_v4.cc)
+target_link_libraries(spectre_v4 safeside)
+
+# build the kernel module
+
+set(MODULE_NAME kernel_data)
```

Comment:
Can you please split the build for the kernel module into a separate CMake module in `third_party/kmod`, then include that module here?

Reply:
Updated.